docs/manual/src/udl/callback_interfaces.md

```diff
@@ -168,3 +153,71 @@ Take care to avoid this by following guidelines:
    references through intermediate objects.
 3. If you need to break the first 2 guidelines, then take steps to manually break the cycles to avoid memory leaks.
    Alternatively, ensure that the number of objects that can ever be created is bounded below some acceptable level.
+
+## Scheduling tasks using the `TaskQueue` callback interface
+
+UniFFI defines the a builtin callback interface, `TaskQueue`,  which can be used to run code using
+the foreign language executor (Kotlin CoroutineScope, Python Executor, Swift DispatchQueue, etc.).
+
+You can then use the `TaskQueue::schedule()` method to run the closure in that background task queue
+and return a future for the result of the closure.
+
+```rust,no_run
+pub struct UserStore {
+    connection: DatabaseConnection,
+    background_queue: uniffi::TaskQueue,
+}
+
+impl UserStore {
+    pub fn new(background_queue: uniffi::TaskQueue) -> Self {
+        Self {
+            connection: DatabaseConnection::new(),
+            background_queue,
+        }
+    }
+
+    pub async fn add_user(&self, username: String, password: String) {
+        await self.background_queue.schedule(move || {
+            self.connection.execute(format!("INSERT INTO users(username, password) VALUES ({username}, {password})"));
+        }
+    }
+
+    pub async fn lookup_password(&self, username: String) -> String {
+        await self.background_queue.schedule(move || {
+            self.connection.execute(format!("SELECT password FROM users WHERE username = {username}"))
+        }
+    }
+
+    pub async fn delete_user(&self, username: String) -> String {
+        await self.background_queue.schedule(move || {
+            self.connection.execute(format!("DELETE FROM users WHERE username = "{username}"));
+        }
+    }
+```
+
+You can also use the `schedule_later<T>(delay: Duration, closure: FnOnce() -> T) -> T` method to run the closure after a delay.
+
+Task queues can be created on the foreign bindings side by passing in the native executer, for example:
+
+```kotlin
+val userStore = UserStore(TaskQueue(Dispatchers.IO))
+```
+
+## Async callbacks are driven by foreign executors
+
+Code that awaits an async callback interface method will be woken up by an FFI call from the UniFFI
+foreign bindings code. This allows your Rust code to be driven by the foreign async executor,
+avoiding the need to spawn threads from Rust or run a `tokio` loop.
+
+See the `user-data-store` example for how this could work in action.
+
+## Fire and forget callbacks
+
+As mentioned above, UniFFI uses the `ForeignFuture<T>` type to handle async callback interface
+returns.  `ForeignFuture` also implements the `fire()` method, which schedules the future to be run
+by the foreign executor, ignoring the result.  This is good for fire-and-forget style callbacks,
+like logging.
```

Comment:
ideally that new example would demonstrate that too, so another "See the example" note seems worthwhile